plutus-use-cases/src/Language/Plutus/Tutorial.lhs

```diff
@@ -0,0 +1,159 @@
+
+and instead define a function that produces the validator script:
+
+> mkValidatorScript :: Campaign -> Validator
+> -- TODO: Rename Validator -> ValidatorScript ??
+> mkValidatorScript campaign = Validator val where
+>   val = UTXO.applyScript inner (UTXO.lifted campaign)
+>   -- ^ val is the obtained by applying `inner` to the lifted `campaign` value
+>   inner = UTXO.fromPlcCode $$(PlutusTx.plutus [|| 
+
+Anything between the `[||` and `||]` quotes is going to be _on-chain code_ and anything outside the quotes is _off-chain code_. We can now implement a lambda function that looks like `mkValidator`, starting with its parameters:
+
+>               \(c :: Campaign) (act :: CampaignAction) (con :: Contributor) (p :: Runtime.PendingTx') ->
+
+Before we check whether `act` is permitted, we define a number of intermediate values that will make the checking code much more readable. These definitions are placed inside a `let` block, which is closed by a corresponding `in` below.
+
+>               let
+>                   infixr 3 &&
+>                   (&&) :: Bool -> Bool -> Bool
+>                   (&&) = $$(Runtime.and)
+
+There is no standard library of functions that are automatically in scope for on-chain code, so we need to import the ones that we want to use from the `Runtime` module using the `\$\$()` splicing operator.
+
+Next, we pattern match on the structure of the `PendingTx'` value `p` to get the runtime information we care about:
+
+>                   Runtime.PendingTx ins outs _ _ (Runtime.Height currentHeight) _ _ = p
+
+This binds `ins` to the list of all inputs of the current transaction, `outs` to the list of all its outputs, and `currentHeight` to the number of blocks (that is, to the current date).
+
+We also need the parameters of the campaign, which we can get by pattern matching on `c`.
+
+>                   Campaign (Runtime.Value target) (Runtime.Height deadline) (Runtime.Height collectionDeadline) campaignOwner = c
+
+Then we compute the total value of all transaction inputs, using `Runtime.foldr` on the list of inputs `ins`.
+
+>                   totalInputs :: Int
+>                   totalInputs =
+>                       let v (Runtime.PendingTxIn _ _ (Runtime.Value vl)) = vl in
+>                       $$(Runtime.foldr) (\i total -> total + v i) 0 ins
+
+We now have all the information we need to check whether the action `act` is allowed. This will be computed as
+
+>                   isValid = case act of
+>                       Refund -> 
+>                           let
+>                               Contributor pkCon = con
+
+In the `Refund` branch we check that the outputs of this transaction all go to the contributor identified by `pkCon`. To that end we define a predicate
+
+>                               contributorTxOut :: Runtime.PendingTxOut -> Bool
+>                               contributorTxOut o = $$(Runtime.maybe) False (\pk -> $$(Runtime.eqPubKey) pk pkCon) ($$(Runtime.pubKeyOutput) o)
```

Comment:
Since we're trying to keep things elementary, I'd avoid use of `maybe` in favour of explicit pattern matching.

Reply:
(Which TBH I somewhat prefer anyway)